File: ch06/two-factor-auth/course-tracker-start/src/main/java/com/manning/sbip/ch06/controller/LoginController.java
package com.manning.sbip.ch06.controller;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;

@Controller
public class LoginController {

    @GetMapping("/login")
    public String login() {
        return "loginNew";
    }

    @GetMapping("/login-error")
    public String loginError(Model model) {
        model.addAttribute("loginError", true);
        return "loginNew";
    }

    @GetMapping("/login-verified")
    public String loginVerified(Model model) {
        model.addAttribute("verified", true);
        return "loginNew";
    }
    
    /* @PostMapping("/login")
	public String loginValidate() {
		return "redirect:/index";
	} */
}

File: ch10/Kotlin/course-tracker-kotlin-app-security/src/main/kotlin/com/manning/sbip/ch10/controller/LoginController.kt
package com.manning.sbip.ch10.controller

import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.ui.set
import org.springframework.web.bind.annotation.GetMapping


@Controller
class LoginController {

    @GetMapping("/login")
    fun login(): String {
        return "login"
    }

    @GetMapping("/login-error")
    fun loginError(model: Model): String {
        model["loginError"] = true
        return "login"
    }
}